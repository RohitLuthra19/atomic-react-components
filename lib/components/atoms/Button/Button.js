// @flow
import * as React from 'react';
import styled from 'styled-components';

import styles from './Button.style';

type Props = {
  className: string,
  children: Node,
  inheritedStyles?: string,
  type?: string,
  primary?: boolean,
  secondary?: boolean,
  elementRef?: any => void,
  disabled?: boolean,
  ariaLabel?: string,
};

const Button = ({
  className,
  children,
  inheritedStyles,
  type,
  primary,
  secondary,
  elementRef,
  disabled,
  ariaLabel,
  ...others
}: Props): Node => (
  <button
    aria-label={ariaLabel || null}
    className={className}
    disabled={disabled}
    ref={elementRef}
    type={type}
    {...others}
  >
    {children}
  </button>
);

Button.defaultProps = {
  inheritedStyles: '',
  type: 'button',
  primary: false,
  secondary: false,
  elementRef: undefined,
  disabled: false,
  ariaLabel: '',
};

export default styled(Button)`${styles}`;
export { Button as ButtonVanilla };
