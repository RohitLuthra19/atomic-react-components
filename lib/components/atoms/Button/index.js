// @flow

export { default } from './Button';
export { ButtonVanilla } from './Button';
export type { Props } from './Button';
