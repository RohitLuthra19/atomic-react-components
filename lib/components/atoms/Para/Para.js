// @flow
import React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';

import styles from './Para.style';

type Props = {
  children: Node,
  className?: string | void,
  inheritedStyles?: string,
  elementLocator?: string,
};

const Para = ({ children, className, inheritedStyles, elementLocator, ...others }: Props): Node => {
  const dataLocator = elementLocator
    ? {
        'data-locator': `para-${elementLocator}`,
      }
    : {};

  return (
    <p className={className} {...dataLocator} {...others}>
      {children}
    </p>
  );
};

Para.defaultProps = {
  inheritedStyles: '',
  className: '',
  elementLocator: '',
};

export default styled(Para)`
  ${styles};
`;
export { Para as ParaVanilla };
