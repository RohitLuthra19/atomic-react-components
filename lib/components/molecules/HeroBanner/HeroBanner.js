// @flow
/**
 *
 * HeroBanner
 *
 */
import React from 'react';
import styled from 'styled-components';
import classnames from 'classnames';
import styles from './HeroBanner.style';
import type { Props } from './types';

const HeroBanner = ({
  className,
  children,
  heroImageWidth,
  childrenContainerWidth,
  renderHeroImage,
  renderExternalContent,
}: Props): Node => {
  return (
    <section className={classnames(className)}>
      <header role="banner">
        {renderHeroImage && (
          <div className="image" heroImageWidth={heroImageWidth}>
            {renderHeroImage()}
          </div>
        )}
        {renderExternalContent && <div className="externalContent">{renderExternalContent()}</div>}
        {children && (
          <div className="children" childrenContainerWidth={childrenContainerWidth}>
            {children}
          </div>
        )}
      </header>
    </section>
  );
};

HeroBanner.defaultProps = {
  className: 'hero',
  heroImageWidth: '50%',
  childrenContainerWidth: '50%',
};

export default styled(HeroBanner)`
  ${styles};
`;

export { HeroBanner as HeroBannerVanilla };
