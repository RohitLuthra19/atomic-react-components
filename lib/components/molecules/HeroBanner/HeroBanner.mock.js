import Image from '../../atoms/Image/Image';

const defaultImage = {
  src: '/images/bg.jpg',
  placeholderSrc: '/images/image-placeholder.png',
  alt: 'Dummy Image',
};

const defaultHeroBanner = {
  renderHeroImage: () => <Image {...defaultImage} />,
  heroImageWidth: '100%',
};

const halfImageHeroBanner = {
  renderHeroImage: () => <Image {...defaultImage} />,
  heroImageWidth: '50%',
  childrenContainerWidth: '50%',
};

const renderExternalContent = {
  renderExternalContent: () => {},
};

export { defaultHeroBanner, halfImageHeroBanner, renderExternalContent };
