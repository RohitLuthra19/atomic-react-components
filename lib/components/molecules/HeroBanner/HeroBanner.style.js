import { css } from 'styled-components';

const styles = css`
  header {
    position: ${props => (props.heroImageWidth === '100%' ? 'relative' : 'static')};
    display: ${props => (props.heroImageWidth === '100%' ? 'block' : 'flex')};
    ${props => (props.renderExternalContent ? '' : props.theme.colors.TOGGLE_SWITCH)};
  }

  .image {
    flex-basis: ${props => (props.heroImageWidth ? props.heroImageWidth : '50%')};
  }

  .children {
    ${props => props.theme.colors.TEXT_DEFAULT};
    text-align: center;
    position: ${props => (props.heroImageWidth === '100%' ? 'absolute' : 'static')};
    ${props => (props.heroImageWidth === '100%' ? 'left:50%' : '')};
    ${props => (props.heroImageWidth === '100%' ? 'top:50%' : '')};
    padding: 10px;
    flex-basis: ${props => (props.childrenContainerWidth ? props.childrenContainerWidth : '50%')};
    transform: ${props => (props.heroImageWidth === '100%' ? 'translate(-50%, -50%)' : '')};
  }

  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;

export default styles;
