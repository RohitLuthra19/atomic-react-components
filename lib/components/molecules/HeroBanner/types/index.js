// @flow
import type { Node } from 'react';

export type Props = {
  children: Node,
  className?: String,
  heroImageWidth?: String,
  childrenContainerWidth?: String,
  inheritedStyles?: string,
  renderHeroImage?: Function,
  renderExternalContent?: Function,
};
