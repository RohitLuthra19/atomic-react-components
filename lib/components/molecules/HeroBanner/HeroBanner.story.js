import React from 'react';
import { storiesOf } from '@storybook/react';
// import { action } from '@storybook/addon-actions';
import { defaultHeroBanner, halfImageHeroBanner, renderHeroCarousel } from './HeroBanner.mock';
import Button from '../../atoms/Button/Button';
import primaryButton from '../../atoms/Button/Button.mock';
import Para from '../../atoms/Para/Para';

// Import Styled Component to showcase variations
import HeroBanner from '.';

storiesOf('Molecules/HeroBanner', module)
  .addParameters({ jest: ['HeroBanner', 'HeroBannerVanilla'] })
  .add('HeroBanner', () => <HeroBanner {...defaultHeroBanner} />)
  .add('HeroBannerWithCTAText', () => (
    <HeroBanner {...defaultHeroBanner}>
      <Para>Content will be displayed here in the banner</Para>
      <Button {...primaryButton}>CTA</Button>
    </HeroBanner>
  ))

  .add('HeroBannerHalf', () => (
    <HeroBanner {...halfImageHeroBanner}>
      <Para>Content will be displayed here in the banner</Para>
      <Button {...primaryButton}>CTA</Button>
    </HeroBanner>
  ))
  .add('HeroCarousel', () => <HeroBanner {...renderHeroCarousel} />);
