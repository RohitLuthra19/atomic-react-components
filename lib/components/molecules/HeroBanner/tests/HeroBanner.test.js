import React from 'react';
import { shallowWithTheme } from '../../../../../__mock__/themeMock';
import { HeroBannerVanilla } from '../index';
import { defaultHeroBanner } from '../HeroBanner.mock';
import Image from '../../../atoms/Image/Image';
import Button from '../../../atoms/Button/Button';
import Para from '../../../atoms/Para/Para';

describe('<HeroBanner />', () => {
  let HeroBannerComponent = '';
  beforeEach(() => {
    HeroBannerComponent = shallowWithTheme(
      <HeroBannerVanilla defaultHeroBanner={defaultHeroBanner} />
    );
  });

  test('should render correctly', () => {
    expect(HeroBannerComponent).toMatchSnapshot();
  });

  test('should render a div', () => {
    expect(HeroBannerComponent.find('section').length).toEqual(1);
  });

  test('should render a header', () => {
    expect(HeroBannerComponent.find('header').length).toEqual(1);
  });

  test('should render a header with role as banner', () => {
    expect(HeroBannerComponent.find('header').prop('role')).toEqual('banner');
  });

  test('should render Hero Image', () => {
    HeroBannerComponent = shallowWithTheme(<HeroBannerVanilla renderHeroImage={() => <Image />} />);
    expect(HeroBannerComponent.find('Image').length).toEqual(1);
  });

  test('should render childen', () => {
    HeroBannerComponent = shallowWithTheme(
      <HeroBannerVanilla defaultHeroBanner={defaultHeroBanner}>
        <Para>Content of the banner will be placed here</Para>
      </HeroBannerVanilla>
    );
    expect(HeroBannerComponent.find('Para').length).toEqual(1);
  });

  test('should render Hero-Carousel', () => {
    /* Since carousel component is not there then for now using Button atom */
    HeroBannerComponent = shallowWithTheme(
      <HeroBannerVanilla renderExternalContent={() => <Button />} />
    );
    expect(HeroBannerComponent.find('Button').length).toEqual(1);
  });
});
