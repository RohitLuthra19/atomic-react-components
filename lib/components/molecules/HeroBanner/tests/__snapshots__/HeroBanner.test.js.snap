// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroBanner /> should render correctly 1`] = `
<section
  className="hero"
>
  <header
    role="banner"
  />
</section>
`;
